catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandlerTest.java

```diff
@@ -0,0 +1,150 @@
+  private static final String QUERY_PARAM = ":transformerId";
+  private static final String RETURN_ID = "kml";
+  private static final String OTHER_RETURN_ID = "xml";
+  private static final String MIME_TYPE = "application/vnd.google-earth.kml+xml";
+  private static final String SAFE_BODY =
+      "{\"src\":\"ddf.distribution\",\"start\":1,\"count\":250,\"cql\":\"anyText ILIKE '*'\",\"sorts\":[{\"attribute\":\"modified\",\"direction\":\"descending\"}],\"id\":\"7a491439-948e-431b-815e-a04f32fecec9\"}";
+  private static final String CONTENT = "test";
+  private static final String SERVICE_NOT_FOUND = "{\"message\":\"Service not found\"}";
+  private static final String SERVICE_SUCCESS = "";
+
+  @Before
+  public void setUp() throws Exception {
+    initMocks(this);
+
+    mockBundleContext = mock(BundleContext.class);
+    mockEndpointUtil = mock(EndpointUtil.class);
+    mockRequest = mock(Request.class);
+    mockResponse = mock(Response.class);
+    mockCqlQueryResponse = mock(CqlQueryResponse.class);
+    mockQueryResponse = mock(QueryResponse.class);
+    mockServiceReference = mock(QueryResponseTransformer.class);
+    mockQueryResponseTransformer = mock(ServiceReference.class);
+    mockServletOutputStream = mock(ServletOutputStream.class);
+    mockHttpServiceResponse = mock(HttpServletResponse.class);
+
+    queryResponseTransformers = new ArrayList<>();
+
+    when(mockQueryResponseTransformer.getProperty(Core.ID)).thenReturn(RETURN_ID);
+
+    MimeType mimeType = new MimeType(MIME_TYPE);
+    mockBinaryContent =
+        new BinaryContentImpl(new ByteArrayInputStream(CONTENT.getBytes()), mimeType);
+
+    queryResponseTransformers.add(mockQueryResponseTransformer);
+
+    cqlTransformHandler = new CqlTransformHandler(queryResponseTransformers, mockBundleContext);
+
+    cqlTransformHandler.setEndpointUtil(mockEndpointUtil);
+
+    when(mockEndpointUtil.safeGetBody(mockRequest)).thenReturn(SAFE_BODY);
+
+    when(mockEndpointUtil.executeCqlQuery(any(CqlRequest.class))).thenReturn(mockCqlQueryResponse);
+
+    when(mockCqlQueryResponse.getQueryResponse()).thenReturn(mockQueryResponse);
+
+    when(mockBundleContext.getService(mockQueryResponseTransformer))
+        .thenReturn(mockServiceReference);
+
+    when(mockServiceReference.transform(mockQueryResponse, Collections.emptyMap()))
+        .thenReturn(mockBinaryContent);
+
+    when(mockResponse.raw()).thenReturn(mockHttpServiceResponse);
+
+    when(mockHttpServiceResponse.getOutputStream()).thenReturn(mockServletOutputStream);
+  }
+
+  @Test
+  public void testNoServiceFound() throws Exception {
+    when(mockRequest.params(QUERY_PARAM)).thenReturn(OTHER_RETURN_ID);
+
+    String res = (String) cqlTransformHandler.handle(mockRequest, mockResponse);
+
+    assertThat(res, is(SERVICE_NOT_FOUND));
+  }
+
+  @Test
+  public void testServiceFoundWithValidResponseAndGzip() throws Exception {
+    when(mockRequest.headers(HttpHeaders.ACCEPT_ENCODING)).thenReturn(GZIP);
```

Comment:
There should be more assertions in these tests. We should check that the handler is attaching the file and that it is setting the correct HTTP headers and status codes.